### ISSUE TITLE
Incorrect typing of iter_chunks in StreamingBody for aiobotocore

### ISSUE THREAD

**Issue by vikigenius** [youtype/mypy_boto3_builder#309]
**Describe the bug**

**To Reproduce**

1. Install `types-aioboto3[essential]`
2. Run `mypy`/`pyright` on the following code sample

```python
import aioboto3


        async with self.session.client('lambda', config=BotoConfig(read_timeout=5)) as lambda_client:
            response = await lambda_client.invoke(
                FunctionName="SOME_NAME",
                InvocationType='RequestResponse',
                Payload=json.dumps(event)
            )
            payload_str: str = ""
            async for chunk in response["Payload"].iter_chunks():  # This complains
                payload_str += chunk.decode('utf-8')
```

**Actual output**

```
Coroutine[Any, Any, Iterator[bytes]] is not iterable
```

**Expected output**
The code actually works. I think the type stubs defined here for iter_chunks is wrong
https://github.com/youtype/mypy_boto3_builder/blob/51f2475433bfed54e73601b342a6aa86fca22515/mypy_boto3_builder/stubs_static/aiobotocore/response.pyi#L32

I think it should return AsyncIterator[bytes] or AsyncGenerator[bytes, None]

Considering this is how the implementation looks

```python
    async def iter_chunks(self, chunk_size=_DEFAULT_CHUNK_SIZE):
        """Return an iterator to yield chunks of chunk_size bytes from the raw
        stream.
        """
        while True:
            current_chunk = await self.read(chunk_size)
            if current_chunk == b"":
                break
            yield current_chunk
```

https://github.com/aio-libs/aiobotocore/blob/5a17c06a0f46faa1de0ee5cf68b3ea4a235ad5d2/aiobotocore/response.py#L107

**Comment by vemel** [comment]
Hello! Thank you for the report.

yes, type annotations for `StreamingBody.iter_chunks` was incorrect. Actually, today I learned something new about typing of async functions with `yield` statement.

```python
from typing import AsyncIterator

async def my_iterator():
   yield 1
   yield 2
   yield 3

# proper type annotation, notice that the function is not async
def my_iterator() -> AsyncIterator[int]
```

I fixed type annotations for `StreamingBody.iter_chunks` and `StreamingBody.iter_lines` in the latest `mypy_boto3_builder` release. Also, I released packages with the fix incuded:

- types-aiobotocore 2.13.1.post1
- types-aiobotocore-lite 2.13.1.post1
- types-aiobotocore 2.13.2.post1
- types-aiobotocore-lite 2.13.2.post1

Please update and let me know if it works as it should.

**Comment by vikigenius** [comment]
This seems to have fixed the issue. Thanks for resolving this so quickly.
